src/OrchardCore.Modules/OrchardCore.OpenId/Configuration/OpenIdServerConfiguration.cs

```diff
@@ -56,142 +42,88 @@ public void Configure(AuthenticationOptions options)
-            options.LogoutEndpointPath = settings.LogoutEndpointPath;
-            options.TokenEndpointPath = settings.TokenEndpointPath;
-            options.UserinfoEndpointPath = settings.UserinfoEndpointPath;
+            if (settings.AuthorizationEndpointPath.HasValue)
+            {
+                options.AuthorizationEndpointUris.Add(new Uri(settings.AuthorizationEndpointPath.Value, UriKind.Relative));
+            }
+            if (settings.LogoutEndpointPath.HasValue)
+            {
+                options.LogoutEndpointUris.Add(new Uri(settings.LogoutEndpointPath.Value, UriKind.Relative));
+            }
+            if (settings.TokenEndpointPath.HasValue)
+            {
+                options.TokenEndpointUris.Add(new Uri(settings.TokenEndpointPath.Value, UriKind.Relative));
+            }
+            if (settings.UserinfoEndpointPath.HasValue)
+            {
+                options.UserinfoEndpointUris.Add(new Uri(settings.UserinfoEndpointPath.Value, UriKind.Relative));
+            }
 
-            options.GrantTypes.Clear();
             options.GrantTypes.UnionWith(settings.GrantTypes);
 
             options.Scopes.Add(OpenIddictConstants.Scopes.Email);
             options.Scopes.Add(OpenIddictConstants.Scopes.Phone);
             options.Scopes.Add(OpenIddictConstants.Scopes.Profile);
-            options.Scopes.Add(OpenIddictConstants.Claims.Roles);
+            options.Scopes.Add(OpenIddictConstants.Scopes.Roles);
         }
 
-        public void Configure(OpenIddictServerOptions options) => Debug.Fail("This infrastructure method shouldn't be called.");
-
-        public void Configure(string name, JwtBearerOptions options)
+        public void Configure(OpenIddictServerDataProtectionOptions options)
         {
-            // Ignore JWT handler instances that don't correspond to the private instance managed by the OpenID module.
-            if (!string.Equals(name, OpenIdConstants.Schemes.Userinfo, StringComparison.Ordinal))
-            {
-                return;
-            }
-
             var settings = GetServerSettingsAsync().GetAwaiter().GetResult();
             if (settings == null)
             {
                 return;
             }
 
-            options.TokenValidationParameters.ValidAudience = OpenIdConstants.Prefixes.Tenant + _shellSettings.Name;
-            options.TokenValidationParameters.IssuerSigningKeys = _serverService.GetSigningKeysAsync().GetAwaiter().GetResult();
-
-            // If an authority was explicitly set in the OpenID server options,
-            // prefer it to the dynamic tenant comparison as it's more efficient.
-            if (settings.Authority != null)
-            {
-                options.TokenValidationParameters.ValidIssuer = settings.Authority.AbsoluteUri;
-            }
-            else
-            {
-                options.TokenValidationParameters.IssuerValidator = (issuer, token, parameters) =>
```

Comment:
None of this gnarly stuff will be required in 3.0, as the OpenIddict validation handler will support JWT and infer the issuer from the request host.